dashboard/modules/job/job_manager.py

```diff
@@ -447,7 +447,16 @@ async def _monitor_job(
 
         while is_alive:
             try:
-                await job_supervisor.ping.remote()
+                # Workaround for https://github.com/ray-project/ray/issues/28058.
+                # Once fixed, replace with `await job_supervisor.ping.remote()`.
+                ref = job_supervisor.ping.remote()
+                not_ready = [ref]
+                while not_ready:
+                    ready, not_ready = ray.wait(ref, timeout=0.1)
+                    if ready:
+                        ray.get(ready)
+                    else:
+                        await asyncio.sleep(0)
                 await asyncio.sleep(self.JOB_MONITOR_LOOP_PERIOD_S)
             except Exception as e:
```

Comment:
Nit: Is it possible to add the specific exception we're trying to detect here? In the old code, it's apparent that the exception would be raise by `await job_supervisor.ping.remote()`, but now it's not immediately apparent what error we're trying to catch.